test/badge.js

```diff
@@ -674,3 +675,225 @@ export async function deleteBadgesOverApi(n) {
     for(let i = 0; i < n; i++)
         await deleteBadgeOverApi();
 }
+
+/**
+ * Downloads the JSON file of a badge from the awarded badge details page.
+ * @returns The content of the json file as string
+ * @param {import('selenium-webdriver').ThenableWebDriver} driver
+ */
+const downloadBadgeJson = async (driver, badgeName) => {
+  const overflowMenu = await driver.findElement(
+    By.css('button:has(svg[icon="icon_more"])')
+  );
+  await overflowMenu.click();
+
+  const downloadButton = await driver.findElement(
+    ExtendedBy.tagWithText('button', `Download JSON-Datei (${currentBadgeStandardVersion})`)
+  );
+  await downloadButton.click();
+
+  await waitForDownload(driver, new RegExp(/^\d{4}-\d{2}-\d{2}-[a-zA-Z0-9_ ]+\.json$/));
+
+  const now = new Date();
+  const badgeJsonName =
+    `${now.getFullYear()}-` +
+    `${('0' + (now.getMonth()+1)).slice(-2)}-` +
+    `${('0' + (now.getDate())).slice(-2)}-` +
+    `${badgeName.replace(' ', '_')}.json`;
```

Comment:
Do we need to have this that way? This means that it could fail if executed at midnight. It's admittedly unlikely, but we could also search for a file matching the regex and read this then.